feat: add progress bar

mmolari opened the pull request:
Added progress bar using indicatif, as suggested by Ivan.

rneher commented:
Nice! The estimate of the remaining time seems to terribly useful though. 
![image](https://github.com/user-attachments/assets/8620400b-e72b-453c-9766-5d3678465b8e)

mmolari commented:
Removed ETA in the progress bar and added the `--no-progress-bar` in the CLI. The bar is printed in stdout and is not printed if the output is piped to a file, only if the output file is specified with the `-o` flag. But in the documentation now we only use the build command with this option.
We could print the progress bar in stderr but I don't think it's good practice...
Other than that I think this is ready for review!

mmolari commented on packages/pangraph/src/commands/build/build_run.rs:
```diff
@@ -74,6 +76,13 @@ pub fn build(fastas: Vec<FastaRecord>, args: &PangraphBuildArgs) -> Result<Pangr
   let tree = build_tree_using_neighbor_joining(graphs)?;
 
+  // Unless no_progress_bar is true, instantiate the progress bar
+  let pb = if args.no_progress_bar || (n_paths <= 1) {
```
for @ivan-aksamentov : do you think this is too verbose? Alternatively I've seen that the progress bar class has an optional progress bar member. One could somehow set this option to None at initialization if `--no-progress-bar` is set. This will also simplify the progress bar update downstream, since the class already take care of progressing the bar only if the option is not None.

mmolari commented on packages/pangraph/src/commands/build/build_run.rs:
```diff
@@ -94,6 +103,10 @@ pub fn build(fastas: Vec<FastaRecord>, args: &PangraphBuildArgs) -> Result<Pangr
 
           clade.data = Some(merge_graphs(left, right, args).wrap_err("When merging graphs")?);
 
+          if let Some(pb) = &pb {
```
Basically this check and the one below could be gone if we just set the option to None... but not sure if it's a good idea...

mmolari replied:
I agree, I'll add this little stylistic modification.

mmolari replied:
Oks I think if you don't see problem with anything should be ready to merge now!
